src/main/java/org/mastodon/mamut/fitting/edgel/SampleEllipsoidEdgel.java

```diff
@@ -118,33 +129,40 @@ public static Ellipsoid sample(
 			}
 			catch ( final IllegalArgumentException e )
 			{
-//				e.printStackTrace();
-//				System.out.println( "oops" );
+				// e.printStackTrace();
+				// System.out.println( "oops" );
 			}
 			catch ( final RuntimeException e )
 			{
-//				System.out.println( "psd" );
+				// e.printStackTrace();
+				// System.out.println( "spd" );
 			}
 		}
 
 		if ( bestEllipsoid == null )
+		{
+			// System.out.println( "no best ellipsoid. potential reason: center of best ellipsoid too far from expected center." );
 			return null;
+		}
 
-		final Ellipsoid refined = fitToInliers( bestEllipsoid, edgels, outsideCutoffDistance, insideCutoffDistance, angleCutoffDistance );
-		if ( refined == null )
+		try
+		{
+			// refined ellipsoid
+			return fitToInliers( bestEllipsoid, edgels, outsideCutoffDistance, insideCutoffDistance,
+					angleCutoffDistance );
+		}
+		catch ( final IllegalArgumentException e )
 		{
-//			System.err.println( "refined ellipsoid == null! This shouldn't happen!");
 			return bestEllipsoid;
 		}
-		return refined;
 	}
 
 	public static Ellipsoid fitToInliers(
 			final Ellipsoid guess,
 			final List< Edgel > edgels,
 			final double outsideCutoffDistance,
 			final double insideCutoffDistance,
-			final double angleCutoffDistance )
+			final double angleCutoffDistance ) throws IllegalArgumentException
```

Comment:
Please remove the unnecessary `throws IllegalArgumentException`. A javadoc that describes what the method does would probably be really helpful. Please describe under which condition the method is expected to throw the IllegalArgumentException.

Reply:
It's ok now